File: Java/shellsort/meta_llama_4_scout_17b_16e_instruc.java
package Java.shellsort;
import java.util.Random;
public class meta_llama_4_scout_17b_16e_instruc {
    
    public static void shellSort(int[] arr) {
        int n = arr.length;
        int gap = n / 2;

        while (gap > 0) {
            for (int i = gap; i < n; i++) {
                int temp = arr[i];
                int j = i;
                while (j >= gap && arr[j - gap] > temp) {
                    arr[j] = arr[j - gap];
                    j -= gap;
                }
                arr[j] = temp;
            }
            gap /= 2;
        }
    }

    public static void main(String[] args) {
        Random rand = new Random();
        int[] arr = new int[1000];
        for (int i = 0; i < 1000; i++) {
            arr[i] = rand.nextInt(10000);
        }

        shellSort(arr);
    }

}

File: Python/shellsort/meta-llama-4-scout-17b-16e-instruc.py
import random

def shell_sort(arr):
    n = len(arr)
    gap = n//2

    while gap > 0:
        for i in range(gap, n):
            temp = arr[i]
            j = i
            while j >= gap and arr[j-gap] > temp:
                arr[j] = arr[j-gap]
                j -= gap
            arr[j] = temp
        gap //= 2

    return arr

# Generate 1000 random inputs
arr = [random.randint(1, 1000) for _ in range(1000)]

# Call the shell_sort function
sorted_arr = shell_sort(arr)